core/src/main/java/nl/nn/adapterframework/jms/PullingJmsListener.java

```diff
@@ -173,7 +173,14 @@ public void closeThread(Map<String,Object> threadContext) throws ListenerExcepti
 
 	@Override
 	public void afterMessageProcessed(PipeLineResult plr, Object rawMessageOrWrapper, Map<String,Object> threadContext) throws ListenerException {
-		String cid = (String) threadContext.get(PipeLineSession.technicalCorrelationIdKey);
+		String replyCid = null;
+
+		if (!isForceMessageIdAsCorrelationId()) {
+			replyCid = (String) threadContext.get(PipeLineSession.correlationIdKey);
+		}
+		if (StringUtils.isEmpty(replyCid)) {
+			replyCid = (String) threadContext.get(PipeLineSession.messageIdKey);
+		}
```

Comment:
Het was misschien toch net iets mooier geweest als dit blokje in de JmsListenerBase had gezeten :)

Reply:
gedaan